src/main/java/org/folio/services/validator/InvoiceValidator.java

```diff
@@ -119,42 +126,76 @@ private void validateInvoiceLineFundDistributions(List<InvoiceLine> invoiceLines
       if (CollectionUtils.isEmpty(line.getFundDistributions())) {
         throw new HttpException(400, FUND_DISTRIBUTIONS_NOT_PRESENT);
       }
-
-      if (isFundDistributionSummaryNotValid(line, currencyUnit)) {
-        throw new HttpException(400, LINE_FUND_DISTRIBUTIONS_SUMMARY_MISMATCH);
-      }
+      validateFundDistributionForInvoiceLine(line.getSubTotal(),line.getFundDistributions(),currencyUnit);
     }
   }
 
-  private boolean isFundDistributionSummaryNotValid(InvoiceLine line, CurrencyUnit currencyUnit) {
-    return ObjectUtils.notEqual(sumMixedDistributions(line, currencyUnit), line.getTotal());
-  }
 
-  private Double sumMixedDistributions(InvoiceLine line, CurrencyUnit currencyUnit) {
-    return sumMixedDistributions(line.getFundDistributions(), line.getTotal(), currencyUnit);
-  }
-
-  private void validateInvoiceAdjustmentsDistributions(List<Adjustment> adjustments, CurrencyUnit currencyUnit) {
-    for (Adjustment adjustment : adjustments){
+  public static void validateInvoiceAdjustmentsDistributions(List<Adjustment> adjustments,CurrencyUnit currencyUnit) {
+    for (Adjustment adjustment : adjustments) {
       if (CollectionUtils.isEmpty(adjustment.getFundDistributions())) {
         throw new HttpException(400, ADJUSTMENT_FUND_DISTRIBUTIONS_NOT_PRESENT);
       }
-
-      if (isFundDistributionSummaryNotValid(adjustment, currencyUnit)) {
-        throw new HttpException(400, ADJUSTMENT_FUND_DISTRIBUTIONS_SUMMARY_MISMATCH);
+      validateFundDistributionForInvoiceLine(adjustment.getValue(), adjustment.getFundDistributions(), currencyUnit);
+    }
+  }
+  public static void validateFundDistributionForInvoiceLine(Double total, List<FundDistribution> fundDistributions,CurrencyUnit currencyUnit) {
```

Comment:
the name of this method implies that it is meant to validate fund distributions for an invoice line, but its parameters do not accepts and invoice line. There ought to be representation in the parameter list that represents the invoice line.

Reply:
yes i agree, it is fixed now.